src/cpuinfo/SDL_cpuinfo.c

```diff
@@ -840,18 +844,40 @@ static const char *SDL_GetCPUName(void)
 int SDL_GetCPUCacheLineSize(void)
 {
     const char *cpuType = SDL_GetCPUType();
+    int cacheline_size;
     int a, b, c, d;
     (void)a;
     (void)b;
     (void)c;
     (void)d;
     if (SDL_strcmp(cpuType, "GenuineIntel") == 0 || SDL_strcmp(cpuType, "CentaurHauls") == 0 || SDL_strcmp(cpuType, "  Shanghai  ") == 0) {
         cpuid(0x00000001, a, b, c, d);
-        return ((b >> 8) & 0xff) * 8;
+	cacheline_size = ((b >> 8) & 0xff) * 8;
+        return cacheline_size;
     } else if (SDL_strcmp(cpuType, "AuthenticAMD") == 0 || SDL_strcmp(cpuType, "HygonGenuine") == 0) {
         cpuid(0x80000005, a, b, c, d);
-        return c & 0xff;
+	cacheline_size = c & 0xff;
+        return cacheline_size;
     } else {
+#if defined(__LINUX__) && defined (__GNUC__)
+        if ((cacheline_size = sysconf(_SC_LEVEL1_DCACHE_LINESIZE)) != -1)
+            return cacheline_size;
+#endif
+#if defined(__LINUX__) /* non glibc linux systems */
+        FILE *f = NULL;
+
+        f = fopen("/sys/devices/system/cpu/cpu0/cache/index0/coherency_line_size", "r");
+        if (f) {
+            fscanf(f, "%d", &cacheline_size);
+            fclose(f);
+            return cacheline_size;
+	}
+#elif defined __FreeBSD__
+#ifdef CACHE_LINE_SIZE
+        cacheline_size = CACHE_LINE_SIZE;
```

Comment:
Is this defined by system headers?

Reply:
yes https://lists.freebsd.org/pipermail/freebsd-arch/2009-April/009190.html

Reply:
Does SDL_cpuinfo.c (maybe indirectly) include the necessary header for it? (which one should it be, sys/param.h?)

Reply:
I did add this line:
```
#if defined (__FreeBSD__)
#include <sys/param.h>
#endif
```

On the commit. If I'm not mistaking that should?

Reply:
Ah, missed that. Sorry for the noise.